service/resource/configmap/resource.go

```diff
@@ -188,7 +191,65 @@ func (r *Resource) GetDeleteState(ctx context.Context, obj, currentState, desire
 }
 
 func (r *Resource) GetUpdateState(ctx context.Context, obj, currentState, desiredState interface{}) (interface{}, interface{}, interface{}, error) {
-	return nil, nil, nil, nil
+	customObject, err := toCustomObject(obj)
+	if err != nil {
+		return nil, nil, nil, microerror.Mask(err)
+	}
+	currentConfigMaps, err := toConfigMaps(currentState)
+	if err != nil {
+		return nil, nil, nil, microerror.Mask(err)
+	}
+	desiredConfigMaps, err := toConfigMaps(desiredState)
+	if err != nil {
+		return nil, nil, nil, microerror.Mask(err)
+	}
+
+	var configMapsToCreate interface{}
+	{
+		configMapsToCreate, err = r.GetCreateState(ctx, obj, currentState, desiredState)
+		if err != nil {
+			return nil, nil, nil, microerror.Mask(err)
+		}
+	}
+
+	var configMapsToDelete []*apiv1.ConfigMap
+	{
+		r.logger.Log("cluster", key.ClusterID(customObject), "debug", "finding out which config maps have to be deleted")
+
+		for _, currentConfigMap := range currentConfigMaps {
+			if !containsConfigMap(desiredConfigMaps, currentConfigMap) {
+				configMapsToDelete = append(configMapsToDelete, currentConfigMap)
+			}
+		}
+
+		r.logger.Log("cluster", key.ClusterID(customObject), "debug", fmt.Sprintf("found %d config maps that have to be deleted", len(configMapsToDelete)))
+	}
+
+	var configMapsToUpdate []*apiv1.ConfigMap
+	if updateallowedcontext.IsUpdateAllowed(ctx) {
```

Comment:
Here we see how we only look for update state in case the update is allowed. All resources have to make sure to only look for the update state in case updates are allowed.

Reply:
Where is this thing set?